### ISSUE TITLE
hull.c fails with assert(num_fields_in_prog <= num_phy_containers)

### ISSUE THREAD

**Issue by taegyunkim** [chipmunk-project/chipmunk#14]
To test [sub.stateful_alu](https://github.com/anirudhSK/chipmunk/blob/master/example_alus/sub.stateful_alu), I ran following command, as [the original domino paper](https://cs.nyu.edu/~anirudh/domino-sigcomm.pdf) says hull.c's most expressive atom is Sub in Table 3.

```
python3 chipmunk.py ../Sketch_from_Domino_progs/hull.sk example_alus/sub.stateful_alu \ 
7 1 codegen hull parallel
```

Btw I'm using hull.sk from [here](https://github.com/aatish17varma/Sketch_from_Domino_progs/blob/master/hull.sk).

I chose 7 and 1 there because the paper also noted that combination. 

What is going wrong here?

**Comment by taegyunkim** [comment]
Add a better error message explaining what is going wrong. 

The reason domino compiler only needed 1 atom per stage was that it had an assumption that each stateless alu could get input from any phv container and it could output to any phv container.

**Comment by taegyunkim** [comment]
Closed by #71
